modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/index/AbstractRocksDbIndexStorage.java

```diff
@@ -221,7 +221,16 @@ String createStorageInfo() {
      *
      * @throws RocksDBException If failed to delete data.
      */
-    public abstract void destroyData(WriteBatch writeBatch) throws RocksDBException;
+    public final void destroyData(WriteBatch writeBatch) throws RocksDBException {
+        destroyData0(writeBatch);
+
+        indexMetaStorage.removeNextRowIdToBuild(writeBatch, indexId, helper.partitionId());
+
+        nextRowIdToBuild = RowId.lowestRowId(helper.partitionId());
```

Comment:
In quite a few places there is that `RowId.lowestRowId(p)`. You always need to know the context to understand that it's not just the 'lowest possible RowId', but that the intention is to refer to the 'default next RowId to build index'. Let's introduce a utility method somewhere that will accept partition ID and compute this expression. This will make the code more comprehensible. The method could be named `defaultNextRowIdToBuildIndex()` - a bit clumsy, but explains the purpose well.